stream/src/main/java/org/apache/kafka/streaming/processor/internals/ProcessorTopology.java

```diff
@@ -27,88 +27,60 @@
 
 public class ProcessorTopology {
 
-    private Map<String, ProcessorNode> processors = new HashMap<>();
-    private Map<String, SourceNode> sourceTopics = new HashMap<>();
-    private Map<String, SinkNode> sinkTopics = new HashMap<>();
+    private List<ProcessorNode> processors;
+    private Map<String, SourceNode> sourceByTopics;
+    private Map<String, SinkNode> sinkByTopics;
 
-    public ProcessorTopology(Map<String, ProcessorNode> processors,
-                             Map<String, SourceNode> sourceTopics,
-                             Map<String, SinkNode> sinkTopics) {
+    public ProcessorTopology(List<ProcessorNode> processors,
+                             Map<String, SourceNode> sourceByTopics,
+                             Map<String, SinkNode> sinkByTopics) {
         this.processors = processors;
-        this.sourceTopics = sourceTopics;
-        this.sinkTopics = sinkTopics;
+        this.sourceByTopics = sourceByTopics;
+        this.sinkByTopics = sinkByTopics;
     }
 
     public Set<String> sourceTopics() {
-        return sourceTopics.keySet();
+        return sourceByTopics.keySet();
     }
 
     public Set<String> sinkTopics() {
-        return sinkTopics.keySet();
+        return sinkByTopics.keySet();
     }
 
     public SourceNode source(String topic) {
-        return sourceTopics.get(topic);
+        return sourceByTopics.get(topic);
     }
 
     public SinkNode sink(String topic) {
-        return sinkTopics.get(topic);
+        return sinkByTopics.get(topic);
     }
 
     public Collection<SourceNode> sources() {
-        return sourceTopics.values();
+        return sourceByTopics.values();
     }
 
     public Collection<SinkNode> sinks() {
-        return sinkTopics.values();
+        return sinkByTopics.values();
     }
 
     /**
      * Initialize the processors following the DAG reverse ordering
```

Comment:
My bad, in an older version of the code I try to do initialization recursively by starting at sink nodes, and call parent.init() first before initializing itself, I later dropped that idea but did not change the comments. Will fix it in #9.